package.json

```diff
@@ -42,6 +42,16 @@
       "types": "./dist/plugins/*.d.ts",
       "require": "./dist/plugins/*.cjs"
     },
+    "./dist/base-plugin.js": {
+      "import": "./dist/base-plugin.esm.js",
```

Comment:
This might backfire because the main bundle will have the same imports but a different copy thereof. So you would basically ship duplicate code, plus the class/type identity will differ. Wavesurfer expects plugins to be subclassesof that specific BasePlugin import, not another copy. Hope this makes sense, it barely does to me :)

Reply:
Yup I think this PR still has some issues beyond just that. Demoted it to a draft again and will look at it some more here in a bit. But I think I might just end up closing it for now.

More to come soon!

Reply:
Looks like it's possible to ignore named exports for CommonJS modules via a rollup plugin.

Another alternative, just to acommodate the CommonJS module, would be to expose BasePlugin as a static property of the WaveSurfer class instead of exporting it as a named export.

Reply:
> This might backfire because the main bundle will have the same imports but a different copy thereof. So you would basically ship duplicate code, plus the class/type identity will differ. Wavesurfer expects plugins to be subclassesof that specific BasePlugin import, not another copy. Hope this makes sense, it barely does to me :)

Just to clarify, this is already an existing problem.  The current exports do not use the same BasePlugin class instance.  Let me give an example.

```js
import RegionsPlugin from 'wavesurfer.js/dist/plugins/regions.esm.js';
import TimelinePlugin from 'wavesurfer.js/dist/plugins/timeline.esm.js';
```

Both of these imports are distinct compilations of the source code with their own cjs/esm/min bundles.  They contain the code for BasePlugin individually and are not referencing the same instance of the class.

This can be illustrated by running the following code:

```js
console.log('same proto?', RegionsPlugin.__proto__ === TImelinePlugin.__proto__);
```

Or using instanceof:

```js
const regions = new RegionsPlugin();
const timeline = new TimelinePlugin();
console.log('instanceof RegionsPlugin.__proto__?', regions instanceof RegionsPlugin.__proto__);
console.log('instanceof TimelinePlugin.__proto__?', regions instanceof TimelinePlugin.__proto__);
```

The way to fix this would be to modify the library to return a single bundle.

```js
import WaveSurfer, { BasePlugin, TimelinePlugin, RegionsPlugin } from 'wavesurfer.js'

// Now TimelinePlugin.__proto__ === RegionsPlugin.__proto__ === BasePlugin
```

You can get this to work pretty easily by having a root index.js that contains the exports.  Use rollup to build one single bundle and export it all.  It lends itself to tree shaking this way, too.  

But it would be a breaking change to people so this would require a major version bump.

Reply:
@katspaugh I updated with your suggestion about adding the classes to WaveSurfer and that is working and seem like a safe change. Whenever you have time to look at this again.

Reply:
True, and thanks for Index.ts pointer.